### DISCUSSION TITLE
How to update the `chains` property on the WalletConnectV2 instance

### DISCUSSION THREAD

**Discussion by Niwilai** [NoahZinsmeister/web3-react#846] [Q&A]
Our Dapp currently supports two chains: Polygon and Celo, which respectively have the following mainnet chain ids:
`[137, 42220]`. I am instantiating the WalletConnect connector as per the example:

```
const chains = [137, 42220]

export const [walletConnectV2, hooks] = initializeConnector<WalletConnectV2>(
  (actions) =>
    new WalletConnectV2({
      actions,
      options: {
        projectId: process.env.NEXT_PUBLIC_WALLET_CONNECT_ID,
        chains,
        optionalChains,
        showQrModal: true,
      },
    })
);
```
Which works just fine. But when trying to connect a (Gnosis)SafeWallet on Celo via WalletConnect, it would not work but instead I am receiving the following error message:

<img width="603" alt="Screenshot 2023-07-01 at 16 31 00" src="https://github.com/Uniswap/web3-react/assets/9334296/4bffc1be-cce5-4db5-9adc-02dbe60a6ab1">

When I only provide one chain in the chains property (`[42220]`) it works fine.

Has anyone come across this before? I would be surprised whether it is really necessary to re-instantiate the connector with a single chain property only two have a new connector to work with SafeWallet.

**Comment by Niwilai** [comment]
When checking [Uniswaps](https://github.com/Uniswap/interface/blob/main/src/connection/index.ts#L95) website, I can see that they are re-instantiating the WC2 connector each time the chain was changed. This seems to be the way forward I guess, so I am closing here.

**Comment by AndreyPetrov74** [comment]
Thank you! Cool info

**Comment by Variant101** [comment]
I have the same question and I'm still can't fix it

**Reply by Niwilai** [reply]
Can you share some code of how you re-creating the instance?
